pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -40,17 +41,64 @@ type VolumeRestrictions struct {
 var _ framework.PreFilterPlugin = &VolumeRestrictions{}
 var _ framework.FilterPlugin = &VolumeRestrictions{}
 var _ framework.EnqueueExtensions = &VolumeRestrictions{}
-
-// Name is the name of the plugin used in the plugin registry and configurations.
-const Name = names.VolumeRestrictions
+var _ framework.StateData = &preFilterState{}
 
 const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = names.VolumeRestrictions
+	// preFilterStateKey is the key in CycleState to VolumeRestrictions pre-computed data for Filtering.
+	// Using the name of the plugin will likely help us avoid collisions with other plugins.
+	preFilterStateKey = "PreFilter" + Name
+
 	// ErrReasonDiskConflict is used for NoDiskConflict predicate error.
 	ErrReasonDiskConflict = "node(s) had no available disk"
 	// ErrReasonReadWriteOncePodConflict is used when a pod is found using the same PVC with the ReadWriteOncePod access mode.
 	ErrReasonReadWriteOncePodConflict = "node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode"
 )
 
+// preFilterState computed at PreFilter and used at Filter.
+type preFilterState struct {
+	// Names of the pod's volumes using the ReadWriteOncePod access mode.
+	readWriteOncePodPVCs sets.Set[string]
+	// The number of references to these ReadWriteOncePod volumes by scheduled pods.
+	numConflictingPVCRefs int
+}
+
+func (s *preFilterState) updateWithPod(podInfo *framework.PodInfo, multiplier int64) {
+	conflicts := s.numConflictingPVCRefsForPod(podInfo)
+	if conflicts > 0 {
+		if multiplier > 0 {
+			s.numConflictingPVCRefs += conflicts
+		} else {
+			s.numConflictingPVCRefs -= conflicts
+		}
+	}
+}
+
+func (s *preFilterState) numConflictingPVCRefsForPod(podInfo *framework.PodInfo) int {
+	var conflicts int
```

Comment:
nit (slightly more idiomatic):
```suggestion
	conflicts := 0
```